### PR TITLE
fix: support skipping updates to the simulation history for clients with disabled history

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

This PR aims to fix an edge case where to support skipping updates to the simulation history for clients with disabled history. 

### `@metamask/transaction-controller`

- **<FIXED>**: Support skipping updates to the simulation history for clients with disabled history

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -3758,6 +3758,7 @@ export class TransactionController extends BaseController<
       {
         transactionId,
         note: 'TransactionController#updateSimulationData - Update simulation data',
+        skipHistory: this.isHistoryDisabled,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:3761]: Rather than duplicating this check here, could we just move the check inside `#updateTransactionInternal` and have it default to use `isHistoryDisabled` if `skipHistory` is `undefined` since it's optional?

This would mean we could centralise the knowledge in one place, and also remove the `skipHistory` property inside `updateTransaction`?

Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts]: That make sense Matthew, updated

**Review by matthewwalsh0 — APPROVED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -3649,7 +3648,7 @@ export class TransactionController extends BaseController<
       updatedTransactionParams =
         this.#checkIfTransactionParamsUpdated(transactionMeta);
 
-      if (skipHistory !== true) {
+      if (!this.isHistoryDisabled && skipHistory !== true) {
```
Comment (dbrans) [inline packages/transaction-controller/src/TransactionController.ts:3651]: Hi @OGPoyraz!

This conditional took me a second to grok, because:
a) expressions like !ishistoryDisabled are a double-negatives.
b) while `skipHistory !== true` is technically correct, it isn't typical javascript, though I understand that it was not part of this PR.

How about something like this:
```suggestion
    const shouldSkipHistory = this.isHistoryDisabled || skipHistory;

    if (!shouldSkipHistory) {
```

Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts]: That make sense Derek, thanks for suggestion.

**Review by dbrans — APPROVED** [review]
